cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/TemplatesConfiguration.java

```diff
@@ -70,6 +70,14 @@ public TemplatesConfiguration(Path configRoot, Trigger trigger, ConfigurationHol
 
         templates = reader.loadTemplates(trigger);
         increments = reader.loadIncrements(templates, trigger);
+        for (Increment inc : increments.values()) {
+            for (Increment incDep : inc.getDependentIncrements()) {
+                for (Template template : incDep.getTemplates()) {
+                    templates.put(template.getName(), template);
+                    inc.getTemplates().add(template);
+                }
+            }
+        }
```

Comment:
the architecture is the following: The templates configration just holds the configuration and the reader reads information and constructs and collects all information. So it would be better, if you can move this code into the reader rather than here to not mix things up.